cloudinit/url_helper.py

```diff
@@ -310,7 +308,7 @@ def _cb(url):
                     filtered_req_args,
                 )
 
-            if session is None:
+            if create_sessions:
```

Comment:
I actually need to come back to this and re-use the session, but kill the subsequent context manager (re-use) which seems to break on 24.04

Reply:
done